servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/RetryingHttpRequesterFilter.java

```diff
@@ -203,7 +203,20 @@ public Completable apply(final int count, final Throwable t) {
                             sdStatus == null ? onHostsAvailable : onHostsAvailable.ambWith(sdStatus), count, t);
                 }
 
-                final BackOffPolicy backOffPolicy = retryFor.apply(requestMetaData, t);
+                final BackOffPolicy backOffPolicy;
+                try {
+                    backOffPolicy = retryFor.apply(requestMetaData, t);
```

Comment:
Technically it shouldn't, but if users supply a buggy Executer, backOffPolicy.newStrategy(executor).apply(offsetCount, t)` can also throw. Should we expand this catch to cover more code inside the current `apply` method?